src/ninja.cc

```diff
@@ -804,6 +809,7 @@ int NinjaMain::ToolInputs(const Options* options, int argc, char* argv[]) {
 "  -0, --print0            Use \\0, instead of \\n as a line terminator.\n"
 "  -E, --no-shell-escape   Do not shell escape the result.\n"
 "  -d, --dependency-order  Sort results by dependency order.\n"
+"  -i, --include-target    Add target before result in every line, separated by the : character.\n"
```

Comment:
If I understand correctly, this is a convenience feature to avoid calling `ninja -t inputs <target>` in a loop, by reusing the same build plan loaded from memory.

However, it *drastically* changes the behavior of this tool as it now performs one graph traversal per target, instead of a single one, and changes the output considerably: in addition to the prefix, actual input paths will be duplicated when they are used by more than one target, which does not happen when not using this flag.

I think the documentation should reflect that *much* more clearly.

Reply:
I thus suggest finding a more descriptive name than `-include-target`. for example `--separate-target-inputs`, wdyt?

Reply:
Finally, this needs a regression test in `misc/output_test.py`. Please add the correspond test cases to `test_tool_inputs()` in that file.

Reply:
Also, are we sure that target names are never going to include a `':` in their path? Maybe use a TAB character as the separator instead, at least these are forbidden from Ninja build plans at the moment. Or allow the user to provide an alternative separator here? Just thinking out loud, let me know what you think.

Reply:
Yeah, a new tool seems to make sense. What about `multi-inputs` to clarify that it provides different sets of inputs?

Regarding flags/features:

- Make `--dependency-order` the default behavior. Historically, this flag was added to `inputs` to make the tool faster (this is important for me too :-)). The initial implementation (before `InputsCollector`) was actually slightly broken and would collect duplicate targets, so `std::sort()` + `std::unique()` calls were needed to get rid of them. I don't see why keeping the output sorted in the new tool would be useful, though you could add a `--sort` option if you really like it.

- Do not implement escaping, but do support `--print0` or `--format=print0` for scripts. It is trivial to implement, and it's a better alternative imho.

Regarding the delimiter, I think TAB is a good default choice, with the ability to use a different one with a flag. Another alternative is to provide a `--format=json` flag to output a JSON object that maps target paths to list of inputs instead (but the result will be significantly slower to generate, and to consume).

Reply:
New PR created https://github.com/ninja-build/ninja/pull/2527